src/backend/dx11/src/lib.rs

```diff
@@ -142,10 +140,15 @@ impl Instance {
         }
     }
 
-    #[cfg(feature = "winit")]
-    pub fn create_surface(&self, window: &winit::window::Window) -> Surface {
-        use winit::platform::windows::WindowExtWindows;
-        self.create_surface_from_hwnd(window.hwnd() as *mut _)
+    pub fn create_surface(
+        &self,
+        has_handle: &impl raw_window_handle::HasRawWindowHandle,
+    ) -> Surface {
+        let hwnd = match has_handle.raw_window_handle() {
+            raw_window_handle::RawWindowHandle::Windows(handle) => handle.hwnd,
+            _ => unreachable!(),
```

Comment:
I think cases like this should be `panic!`, because they will easily be reachable once the enum gets new variants

Reply:
Yes of course, I hadn't considered that. Do you have any suggestions for the errors message(s)?

Reply:
Yes of course, I hadn't considered that. Do you have any suggestions for the error message(s)?

Reply:
It actually sounds like we need an error code returned for that mismatch?..

Reply:
> It actually sounds like we need an error code returned for that mismatch?..

Could you explain that a bit further, please?

Reply:
Just adding something like `struct InitError` in `window.rs` module, so that all `create_surface_from_raw` calls would return `Result<Surface, window::InitError>` instead of panicking.

Reply:
I thought that was what you were implying, just wanted to double check :)